app/posts/releasing-rails-libraries-for-version-5.md

```diff
@@ -0,0 +1,123 @@
+
+Task lists are intended to help users manage many tasks, especially when the user:
+
+- might not be able to do them all in one sitting
+- is able to complete them in any order
+
+Like with the other components, the Rails implementation borrows heavily from the GOV.UK design system reference, but translates it into an syntax that will be immediately familiar to Ruby on Rails developers.
+
+```erb
+<h2 class="govuk-heading-m">Confirm your identity</h2>
+
+<%=
+  govuk_task_list(id_prefix: "about-you") do |task_list|
+    task_list.with_item do |item|
+      item.with_title(text: "Personal details")
+      item.with_status do
+        govuk_tag(text: "Complete", colour: "green")
+      end
+    end
+    task_list.with_item do |item|
+      item.with_title(text: "Contact details")
+      item.with_status do
+        govuk_tag(text: "Not started yet", colour: "Red")
+      end
+    end
+    task_list.with_item(cannot_start_yet: true) do |item|
+      item.with_title(
+        text: "Identity check",
+        hint: <<~HINT
+          We need your personal and contact details in order
+          to begin confirming your identity
+        HINT
+      )
+      item.with_status(text: "Cannot start")
+    end
+  end
+%>
+```
+
+The above snippet will render this task list:
+
+![A GOV.UK design system task list with entries for personal and contact details, and a greyed out identity check task that cannot yet be started](/assets/posts/releasing-rails-libraries-for-version-5/rendered-task-list.png)
+
+## New link and button helpers
+
+The govuk-components link and button helpers are probably the most widely used feature offered by the gem.
+
+Their role is to take `some text` and `#a-hyperlink` and generate a link with design system classes like `govuk-link`, so:
+
+```erb
+<%= govuk_link_to("some text", "#a-hyperlink") %>
+
+<!-- or -->
+
+<%= govuk_link_to("#a-hyperlink") do %>
+  some text
+<% end %>
+
+<!-- becomes -->
+
+<a class="govuk-link" href="#a-hyperlink">some text</a>
+```
+
+In November 2022 [I tried to add a new feature](https://github.com/x-govuk/govuk-components/pull/363) to `govuk_link_to` which allowed developers to call it with `new_tab: true`, and the resulting link would open pages in a new tab.
+
+Unfortunately [I had to revert the change shortly afterwards](https://github.com/x-govuk/govuk-components/pull/399) because mixing new positional keyword arguments, which were added to Ruby in 2019 with Rails' link helpers, which haven't really changed since 2004, caused lots of problems. Thankfully we uncovered these in our pre-release testing so didn't break anything important.
```

Comment:
I found this paragraph slightly confusing - do you mean "mixing keyword arguments _with_ positional arguments"?

Reply:
Yeah this one's a nightmare, will redo